.github/PULL_REQUEST_TEMPLATE.md

```diff
@@ -0,0 +1,26 @@
+<!-- Thank you for contributing! -->
+<!-- Please fill this template before submitting your PR (fill the boxes using "x") -->
+
+**Your checklist for this pull request**
+- [ ] I've read the [contributing guideline](CONTRIBUTING.md).
+- [ ] I've tested my changes by building and running the project, and testing changed functionality (if applicable)
+- [ ] I've added automated tests for my change (if applicable, optional)
+- [ ] I've updated documentation to reflect my change (if applicable)
+
+**What is the current behaviour?**
+<!-- Explain how the code works currently -->
+
+**What is the new behaviour?**
+<!-- Explain how the code works after your changes -->
+
+**Test plan**
+<!-- Explain how to test your changes -->
+
+<!-- After submitting, your code will be tested by the CI pipeline. Please
+ensure that all tests pass. If they don't at first, please update your code -->
+
+**Closing issues**
+
+<!-- Add in issue numbers related to this PR, if applicable -->
+
+fixes #issuenumber
```

Comment:
While I really like the rest of the templates, I think the PR template could be improved. I think it's overcomplicated for now.

The checklist is unclear due to these `(if applicable)`. If it doesn't apply to me, should I check the box or leave it unchecked? The questions about "current/new behaviour" could be merged into "what behaviour is changed and how", the test plan IMHO could be skipped if the PR adds appropriate tests to the CI - the code would speak for itself, such question is only applicable if some feature is not that easy to test with CI or sb doesn't want to implement tests right now and would stay with manual testing

Reply:
I believe that Test plans aren't only technical tests, but mostly instructions for the reviewer what do and how to check that the feature is solid and didn't break anything. Including presentation of edge cases.

Examples:
big feature - detailed and thorough test plan - https://github.com/radareorg/cutter/pull/2172
small UI addition - on point test plan - https://github.com/radareorg/cutter/pull/2211

Simples example from mquery - https://github.com/CERT-Polska/mquery/pull/176